webui/react/src/stores/users.tsx

```diff
@@ -29,43 +39,57 @@ type FetchUsersConfig = {
 const UsersContext = createContext<UsersContext | null>(null);
 
 export const UsersProvider: React.FC<PropsWithChildren> = ({ children }) => {
-  const [users, setUsers] = useState<Loadable<DetailedUser[]>>(NotLoaded);
-  const [currentUser, setCurrentUser] = useState<Loadable<DetailedUser>>(NotLoaded);
+  const [usersByKey, setUsersByKey] = useState<Map<string, UsersPagination>>(
+    Map<string, UsersPagination>(),
+  );
+  const [users, setUsers] = useState<Map<number, DetailedUser>>(Map<number, DetailedUser>());
+  const [currentUser, setCurrentUser] = useState<Loadable<number>>(NotLoaded);
```

Comment:
Does it represent userId? if so, this should be `currentUserId` instead of `currentUser`?